src/git-sync

```diff
@@ -26,7 +29,11 @@ function preconditions {
   if [ "$1" = "--all" ]; then
     sync_all=true
   elif [ "$(is_feature_branch "$INITIAL_BRANCH_NAME")" = false ]; then
+    # syncing a non-feature branch
```

Comment:
This method is getting a bit unwieldy, and in my opinion the comments add some high-level guidance that isn't immediately obvious without deeply thinking about the line above.